agenta-cli/agenta/sdk/agenta_tracing.py

```diff
@@ -0,0 +1,106 @@
+# Stdlib Imports
+import uuid
+from datetime import datetime
+from typing import Optional, Dict, Any, Optional
+
+# Own Imports
+from agenta.client.backend import client
+from agenta.client.backend.types.span_status import SpanStatus
+from agenta.client.backend.client import AsyncObservabilityClient
+
+
+class LLMTracing:
+    def __init__(self, base_url: str, api_key: Optional[str] = None):
```

Comment:
On how to make the tracing async, check the implementation of langsmith. 

https://github.com/langchain-ai/langsmith-sdk/blob/05ea3dfad9882bdb687bc0368f04aee4b3f9afbf/python/langsmith/client.py#L363

Basically they create a thread with a queue, then send the trace and span requests to the queue and have that thread manage these

Reply:
Thank you! I'll have a look.